admin/yum.py

```diff
@@ -0,0 +1,175 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Effectful interface to RPM tools.
+"""
+
+import os
+
+import requests
+from requests_file import FileAdapter
+from characteristic import attributes
+from effect import sync_performer, TypeDispatcher
+from subprocess import check_call
+
+
+@attributes([
+    "source_repo",
+    "target_path",
+    "packages",
+    "flocker_version",
+    "distro_name",
+    "distro_version",
+])
+class DownloadPackagesFromRepository(object):
+    """
+    Download a given set of RPMs from a repository.
+
+    :ivar bytes source_repo: Location of repository.
+    :ivar FilePath target_path: Directory to download packages to.
+    :ivar list packages: List of bytes, package names to download.
+    :param bytes flocker_version: The version of flocker to download packages
+        for.
+    :param distro_name: The name of the distribution to download packages for.
+    :param distro_version: The distro_version of the distribution to download
+        packages for.
+    """
+
+
+@sync_performer
+def perform_download_packages_from_repository(dispatcher, intent):
+    """
+    See :class:`DownloadPackagesFromRepository`.
+    """
+    # TODO avoid circular imports when importing these somewhere else
+    # Reviewer - before I do, this might be easier if the logic to get
```

Comment:
https://clusterhq.atlassian.net/browse/FLOC-1223 suggests moving into `flocker`, so probably just link to that here.